### ISSUE TITLE
2022.2 support?

### ISSUE THREAD

**Issue by AnttiLukats** [OVGN/OpenHBMC#11]
the current version does not work with Vivado 2022.2 steps to reproduce:
take GIT files, create project in 2020.2, change constraints compile
test with real hardware CR00107 all working
open project in 2022.2, upgrade IP
run the flow
Error running propagate TCL procedure......................

Any chances this issue will be resolved in near future? would be nice to see support in recent release of Vivado!

UPDATE: the issue seems to be related to ADDRESS MAP. If OpenHBMC is "unassigned" in memory map view then all runs to bitgen without errors, well the design will not work, but it emits also no error. Assigning the memory regions will trigger error and the memory regions will be unassigned again automatically.

**Comment by OVGN** [comment]
Yes, I'm going to create several sets of sources to support different Vivado versions. HDL sources are going to be the same, but looks like I have to fix something in .tcl or .xml files.

**Comment by AnttiLukats** [comment]
Yes, I think it is something in the TCL portion of the core,if lucky you can create a core that is compatible with both versions.

**Comment by AnttiLukats** [comment]
Any good news on the new version update?

**Comment by hiroyukikawai86** [comment]
Hello,

I also face with the same problem. Would be really appreciated if you could fix this issue.
I have also tried to solve the issue, though.

**Comment by mohammadhgh** [comment]
I have been using OpenHBMC on Vivado 2022.2 and 2023.1. The problem is in bd.tcl file as @AnttiLukats mentioned. A temporary workaround is to comment out the parts of the [OpenHBMC/bd/bd.tcl](https://github.com/OVGN/OpenHBMC/blob/fc1c154877d5ac5100d97368e6e656c96e0d6a70/OpenHBMC/bd/bd.tcl) file that does the address propagation. This solves the problem but address widths need to be handled manually. I have attached a patch for changes that I did.
[bd_tcl.patch](https://github.com/OVGN/OpenHBMC/files/12387749/bd_tcl.patch.txt)
